src/meta/src/manager/catalog/mod.rs

```diff
@@ -833,7 +832,36 @@ impl CatalogManager {
     }
 
     /// This is used for both `CREATE TABLE` and `CREATE MATERIALIZED VIEW`.
-    pub async fn start_create_table_procedure(
+    pub async fn start_create_table_procedure(&self, table: &Table) -> MetaResult<()> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        let user_core = &mut core.user;
+        database_core.ensure_database_id(table.database_id)?;
+        database_core.ensure_schema_id(table.schema_id)?;
+        for dependent_id in &table.dependent_relations {
+            database_core.ensure_table_view_or_source_id(dependent_id)?;
+        }
+        #[cfg(not(test))]
+        user_core.ensure_user_id(table.owner)?;
+        let key = (table.database_id, table.schema_id, table.name.clone());
+
+        database_core.check_relation_name_duplicated(&key)?;
+
+        if database_core.has_in_progress_creation(&key) {
+            bail!("table is in creating procedure");
+        } else {
+            database_core.mark_creating(&key);
+            database_core.mark_creating_streaming_job(table.id, key);
+            for &dependent_relation_id in &table.dependent_relations {
+                database_core.increase_relation_ref_count(dependent_relation_id);
+            }
+            user_core.increase_ref(table.owner);
+            Ok(())
+        }
+    }
+
+    /// This is used for both `CREATE TABLE` and `CREATE MATERIALIZED VIEW`.
```

Comment:
Need to update the docs for these functions.

Reply:
Done.